types/chartjs-plugin-doughnutlabel-rebourne/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// Type definitions for chartjs-plugin-doughnutlabel-rebourne 3.0
+// Project: https://github.com/alexkuc/chartjs-plugin-doughnutlabel-rebourne
+// Definitions by: Piotr Ładoński <https://github.com/Voodu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.1
+
+import { Chart, ChartType, Plugin } from 'chart.js';
+import { EmptyObject } from 'chart.js/types/basic';
+
+declare module 'chart.js' {
+    interface PluginOptionsByType<TType extends ChartType> {
+        doughnutlabel?: DoughnutLabel.Options;
+    }
+}
+
+declare namespace DoughnutLabel {
+    interface Options {
+        labels?: Label[];
+        /** Which plugin core api to use to draw labels */
+        api?: Exclude<keyof Plugin, 'id'>;
+        /** How much padding to add when scaling very large text (value in percentage of 100) */
+        paddingPercentage?: number | `${number}`;
+        /** Show labels or not (plugin scope) */
+        display?: boolean;
+        /** Color of the labels (plugin scope) */
+        color?: CanvasRenderingContext2D['fillStyle'];
+        /** The font options used to draw the label text (plugin scope) */
+        font?: Font;
+    }
+
+    interface Label {
+        /** The text to display */
+        text: string;
+        /** The font options used to draw the label text (single label scope) */
+        font?: Font;
+        /** Color of the labels (single label scope) */
+        color?: CanvasRenderingContext2D['fillStyle'];
+        /** Show the label or not (single label scope) */
+        display?: boolean;
+    }
+
+    interface Font {
+        /** Defaults to `Chart.defaults.global.defaultFontFamily` */
+        family?: string;
+        /** Defaults to 1.2 */
+        lineHeight?: number | `${number}`;
+        /** Defaults to `Chart.defaults.global.defaultFontSize` */
+        size?: number | `${number}`;
+        /** Defaults to `Chart.defaults.global.defaultFontStyle` */
+        style?: string;
+        /** Defaults to `'normal'` */
+        weight?: string;
+    }
+
+    interface Internal {
+        _drawLabels: (chart: Chart, args: EmptyObject, options: Options) => void;
```

Comment:
where does this api come from?
https://github.com/alexkuc/chartjs-plugin-doughnutlabel-rebourne/search?q=_drawLabels

Reply:
It can be found here https://github.com/alexkuc/chartjs-plugin-doughnutlabel-rebourne/blob/master/src/plugin.js
I wasn't sure if that should be in the typing, as it seems to be some internal method, but it's visible, for example, when logging the whole object to the console. It doesn't seem like it should be used by itself in any situation.

As it's the first time I write types, I'd be grateful for the advice, on how such 'private' methods/props should be treated.

Reply:
Ok, I think it doesn't make sense to expose props that shouldn't be used. I removed the whole `Internal` interface and related imports